<?xml version="1.0" encoding="utf-8" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
     <head>
<!-- automatically generated by Darwin
prepared on Mon Feb  1 10:20:01 CET 2016
running on linneus78
by user adriaal
-->
<title>OMA stand-alone</title>
<link href="../browser.css" rel="stylesheet" type="text/css"></link>
<link href="manual.css" rel="stylesheet" type="text/css"></link>
</head><body >
<script type="text/javascript">var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
   document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));</script><script type="text/javascript">try {
       var pageTracker = _gat._getTracker("UA-1093824-1");
        pageTracker._trackPageview();
	 } catch(err) {}</script><div class="content"><div class="title">OMA stand-alone</div><div class="navigation"><a href="#introduction">Introduction</a> - <a href="#downloads">Downloads</a> - <a href="#installation">Installation</a> - <a href="#usage">Usage</a> - <a href="#fileformats">File formats</a> - <a href="#parameters">Parameters</a> - <a href="#gettinghelp">Getting help</a> - <a href="#license">License</a> - <a href="http://omabrowser.org">Omabrowser</a></div><a name="introduction"><h1>Introduction</h1>
</a><p>You can run OMA as a stand-alone program.
            Included are the algorithms for OMA itself plus its 
            addition ESPRIT. The software can be installed on Linux (x86, 
            both 64bit and 32bit) and MacOSX (x86, both 32bit and 64bit).</p>
<p>For a short summary and pointers to detailed algorithmic 
            publications of OMA, please refer to the OMA browser page:</p>
<p><a href="http://omabrowser.org/oma/about/">http://omabrowser.org/oma/about/</a></p>
<p>For background info on ESPRIT, 
            please refer to this article:</p>
<p><a href="http://dx.doi.org/10.1093/bib/bbr038">http://dx.doi.org/10.1093/bib/bbr038</a></p>
<p>If you have specific questions about the installation 
            or the usage of OMA, please contact 
            adrian.altenhoff <i>at</i>
 inf.ethz.ch or christophe.dessimoz <i>at</i>
 unil.ch.</p>
<a name="downloads"><h1>Downloads</h1>
</a><p>The current version of OMA stand-alone can be 
                found here:</p>
<p><a href="http://omabrowser.org/standalone/OMA.1.0.5.tgz">OMA.1.0.5.tgz</a></p>
<a name="installation"><h1>Installation</h1>
</a><p>You do not need to install OMA stand-alone on your
            system; the script will also run if you just call it by using 
            the complete path to <tt>bin/oma</tt>
 in the installer folder. 
            But we still encourage you to run the installer script, since it 
            makes working with OMA considerably more convenient.</p>
<p>To install OMA stand-alone on your system, download
            the installer, untar the package and run the included installer
            script:</p>
<p><pre>curl http://omabrowser.org/standalone/OMA.1.0.5.tgz -o oma.tgz
tar xvzf oma.tgz
cd OMA.1.0.5
./install.sh /your/install/prefix</pre>
</p>
<p>If you do not choose an install prefix, OMA will be 
            installed in <tt>/usr/local/OMA</tt>
 (for this, you might need 
            to install it using the root account or <tt>sudo</tt>
).</p>
<p>After installation, make sure the <tt>bin</tt>
 folder of OMA is in your <tt>PATH</tt>
 variable, e.g., if
            you are using bash and used <tt>/your/install/prefix</tt>
 as
            installer prefix, add a line in <tt>~/.profile</tt>
 such as:</p>
<p><pre>export PATH=$PATH:/your/install/prefix/OMA/bin</pre>
</p>
<p>For other shells, choose the appropriate syntax.</p>
<a name="usage"><h1>Usage</h1>
</a><p>First, set up a working directory. Copy the file <tt>parameters.drw</tt>
 into this folder and change it to your needs. 
        Create a directory <tt>DB</tt>
 in your working directory that holds 
        the genome data in FASTA format (see "File formats") and copy your 
        data into this directory. If you want to use ESPRIT, the FASTA file 
        containing the contigs should be called <tt>{YourGenome}.contig.fa</tt>
. Then, simply call <tt>OMA</tt>
 from your working directory to 
        run OMA and/or ESPRIT</p>
<p>If you have not installed OMA yet, use the complete
        path to <tt>bin/oma</tt>
 in the installer folder to start the
        script.</p>
<p>As an example, assume you installed <tt>OMA</tt>
 in <tt>/your/install/prefix</tt>
 and want to use ESPRIT on two genome 
        files and one file with contigs (all in <tt>/home/you/fasta</tt>
, do something like this:</p>
<p><pre># create working directory
mkdir myWorkingDir
cd myWorkingDir
# create DB directory in working directory
mkdir DB
# copy FASTA files into DB directory
cp /home/you/fasta/yourFirstGenomeFile.fa DB/
cp /home/you/fasta/yourSecondGenomeFile.fa DB/
cp /home/you/fasta/yourContigFile.contig.fa DB/
cp /your/install/prefix/OMA/OMA.1.0.5/parameters.drw ./
# adjust parameters
vim parameters.drw
# run OMA
OMA</pre>
</p>
<p>To get a first impression of <tt>OMA</tt>
 you 
        could <tt>cd</tt>
 into the <tt>ToyExample</tt>
 directory, 
        have a look at <tt>parameters.drw</tt>
 and run <tt>OMA</tt>
 to process our example files.</p>
<a name="parallelization"><h2>Parallelization</h2>
</a><p>The all-against-all phase of OMA is the most time-consuming one, but it can be parallelized (unlike all other steps, which cannot run in parallel). The way it works is that the parameter "AlignBatchSize" and the total number of genomes (n) will determine into how many chunks the all-against-all phase is divided. AlignBatchSize will split the n*(n-1)/2 genome pairs further into chunks of at most "AlignBatchSize" alignments. The larger AlignBatchSize is, the more and smaller jobs will be executed.</p>
<p>Scheduling is straightforward: all compute 
        processes need to start from the same directory, and each one will try to do an equal 
        amount of chunks sequentially. However, before starting a new chunk, each process ensures that it has not yet already been processed by another process (i.e. no result file yet exists).</p>
<p>On a single computer with multiple processors and/or cores, it is recomended to start "N" parallel processes with the "-n" option, i.e.<pre>OMA -n 5</pre>
 will start 5 parallel jobs. Note that on HPC with schedulers there is a better way described <a href="#schedulers">below</a></p>
<p>Therefore, there is not need to specify which parts are 
        to be done by which process. One should only ensure that all processes 
        start from a shared directory, such that each chunk gets 
        computed by a single process only.</p>
<a name="schedulers"><h3>Parallelization with LSF, PBS Pro, Slurm, or SunGridEngine</h3>
</a><p>With a scheduler such as LSF or SGE, running parallel jobs is particularly easy,
        as the parallel jobs can be start using as a job array. OMA will 
        automatically spread the work for the all-against-all among all processes.
        For a brief discussion on the required resources refer to the section <a href="#resources">below</a>
        Do not start OMA with the -n option for that. Instead, use a job 
        array with one of the supported schedulers below (the example is to 
        start 100 jobs in parallel)</p>
<a name="LSF"><h3>LSF</h3>
</a><p><pre>bsub -J &quot;OMA[1-100]&quot; -o &quot;out.%I&quot; &quot;OMA&quot;</pre>
</p>
<a name="SGE"><h3>Sun Grid Engine (aka Oracle Grid Engine)</h3>
</a><p><pre>qsub -t 1-100 &quot;OMA&quot;</pre>
</p>
<a name="PBSPro"><h3>PBS Pro</h3>
</a><p>Prepare a job script called e.g. job.sh:<pre>
#!/bin/bash
# set the number of nodes and processes per node
#PBS -l select=1:ncpus=1:mem=1000mb
# set max wallclock time
#PBS -l walltime=01:00:00
#PBS -J 1-100
OMA</pre>
The script can then be submitted as follows:<pre>qsub job.sh</pre>
</p>
<a name="Slurm"><h3>Slurm</h3>
</a><p><pre>sbatch --array=1-100 -N1 &lt;&lt;EOF
#!/bin/sh 
/absolute/path/to/bin/OMA
EOF</pre>
</p>
<a name="resources"><h2>Required Resources</h2>
</a><p>Depending on the dataset to be analysed, OMA can require
        quite a significant amount of computational resources, i.e. RAM and cpu
        time. Most computing time is spent to compute the all-against-all
        sequence alignments, which is why this part has been parallelized.
        Although cpu intensive this phase does not require too much memory. As
        a rule of thumb you can assume it requires roughly 10 times the size of
        your DB folder, but at minimum 650MB. The second part of OMA runs sequential on a single
        core, but it requires a lot more memory: Again, as a rule of thumb,
        count on <tt>400MB * pow(nr_genomes, 1.4)</tt>
. Obviously this
        depends also a lot on the size of your genomes. 60 metazoas have been
        successfully computed using 120GB, and for the same number of bacterial
        genomes, 50GB were reported to be enough.</p>
<p>Because of this imbalance regarding the required memory
        the OMA starter script has command line option <tt>-s</tt>
 to run
        only the first part of the OMA pipeline. Using this option, the
        computation can be split into a parallel phase with little memory 
        requirement , and a single process requiring a lot more RAM. Running 
        the OMA thus becomes a two stage process like this:<pre>
# first stage, little memory:
OMA -s 

# once first stage terminated, run second stage
OMA</pre>
Note that this staging is mostly useful on computing clusters where 
        often memory and cpu time has to be reserved at job submission.</p>
<a name="Adding/Updating new genomes"><h2>Adding/Updating new genomes</h2>
</a><p>It is possible to add new genomes without recomputing 
        the all-against-all phase for pre-existing genomes. To do so, simply
        add the new fasta databases in the DB/ directory and re-run OMA.
        Likewise, it is possible to update a genome by deleting the old genome
        from the DB/ directory, and adding a new file. <b>Important: </b>
to avoid clashes with previously computed results, the updated genome 
        must use a different filename than the any previously computed 
        result.</p>
<a name="fileformats"><h1>File Formats</h1>
</a><a name="iformat"><h2>Input Files</h2>
</a><p>OMA uses two different input formats: FASTA files for 
            genome input and a Darwin file for parameter input.</p>
<p>The Fasta format is explained in detail on <a href="http://en.wikipedia.org/wiki/FASTA_format">wikipedia</a>.</p>
<p>As almost everywhere else, OMA uses the greater-than 
            symbol ">" to distinguish labels from sequences. Each sequence 
            in a genome is supposed to have its own label. Have a look at the 
            FASTA files included in ToyExample/DB in our installer package for 
            some example files.</p>
<p>In case your genomes contain multiple alternative 
            splicing variants, you can add a text-based file per genome called<tt>DB/{YourGenome}.splice</tt>
that put together the different
            splicing variants, e.g. to indicate that the three splicing variants
            ENSP00000384207, ENSP00000263741 and ENSP00000353094 are encoded by the 
            same gene, add the following line to the splice file:</p>
<p><pre>ENSP00000384207; ENSP00000263741; ENSP00000353094</pre>
OMA requires that the individual IDs are uniq prefixes of your 
            FASTA headers.</p>
<p>If you want to use ESPRIT, make sure that FASTA files
            containing contigs are called <tt>{YourGenome}.contig.fa</tt>
.
            So if you want to experiment with some mouse genome, call the 
            FASTA file <tt>mouse.contig.fa</tt>
 or <tt>mymouse.contig.fa</tt>
 or something similar.</p>
<p>Parameter files use Darwin syntax.  Key-value-pairs 
            are written as</p>
<p><pre>key := value;</pre>
</p>
<p>Note the colon in <tt>:=</tt>
 and the semicolon at 
            the end of the line. If your parameter file does not use valid 
            Darwin syntax, OMA will print out a short message and stop its 
            execution.</p>
<a name="oformat"><h2>Output Files</h2>
</a><a name="omaout"><h3>OMA Output</h3>
</a><p>The output of OMA gets written to files stored in 
            a directory <tt>Output</tt>
 in your working directory. This can be 
            changed by changing the <tt>OutputFolder</tt>
 parameter, There are 
            text files and directories organized as described in Table 1.</p>
<table><caption>Table 1: Contents of the OMA output files</caption><thead>
<tr><th align="left"><b>Filename or Directory</b>
</th><th align="left"><b>Contents</b>
</th></tr>
</thead>
<tbody><tr><td colspan=2 style="padding: 0px; border-bottom:1px solid black"></td></tr>
<tr><td align="left"><tt>Map-SeqNum-ID.txt</tt>
</td><td align="left"><p>Lists all genes of all datasets  
                    with their unique sequence number and the labels read from
                    the FASTA files.</p>
</td></tr>
<tr><td align="left"><tt>OrthologousGroups.txt</tt>
</td><td align="left"><p>The groups of orthologs are
                    given as one per row, starting with a unique group
                    identifier, followed by all group members, all separated
                    by tabs.</p>
</td></tr>
<tr><td align="left"><tt>OrthologousMatrix.txt</tt>
</td><td align="left"><p>More compact version of <tt>OrthologousGroups.txt</tt>
. The groups of orthologs
                    are given as matrix with group per row and one genome
                    per tab-separated column. Numbers refer to entry number as
                    listed in the file <tt>Map-SeqNum-ID.txt</tt>
.</p>
</td></tr>
<tr><td align="left"><tt>OrthologousGroups.orthoxml</tt>
</td><td align="left"><p>The OMA groups of orthologs stored in <a href="http://orthoxml.org">orthoxml format</a>.</p>
</td></tr>
<tr><td align="left"><tt>OrthologousGroupsFasta/</tt>
</td><td align="left"><p>Each OMA group is provided as a separate
                    Fasta file, with the species name as identifer. This
                    format is particularly suitable as starting point for
                    a phylogenetic reconstruction.</p>
</td></tr>
<tr><td align="left"><tt>PairwiseOrthologs/</tt>
</td><td align="left"><p>The textfiles in <tt>Output/PairwiseOrthologs</tt>
 are named according to <tt>{genome a}-{genome b}.txt</tt>
 and consist of a list of pairwise orthologs for the two given
                    genomes. Every pair is listed only once, and in no particular
                    order. Each line in the file contains one pair; all fields are 
                    separated by tabs. In the first two field, the unique IDs of the
                    proteins are given. The next two fields contain the labels of
                    the proteins, and in the last two fields, the type of orthology
                    and (if any) the OMA group is given.</p>
</td></tr>
<tr><td align="left"><tt>OrthologousPairs.orthoxml</tt>
</td><td align="left"><p>The pairwise orthologs stored in <a href="http://orthoxml.org">orthoxml format</a>. Each group in the file will have
                          orthologs between genes from only two genomes.</p>
</td></tr>
<tr><td align="left"><tt>HierarchicalGroups.orthoxml</tt>
</td><td align="left"><p>The hierarchical groups of orthologs
                    in OrthoXML format. A detailed description of how
                    these groups are computed is forthcoming.</p>
</td></tr>
<tr><td align="left"><tt>EstimatedSpeciesTree.nwk</tt>
</td><td align="left"><p>The inferred species tree on which 
                    the hierarchical groups inference procedure is based,
                    in Newick format.</p>
</td></tr>
<tr><td align="left"><tt>HOGFasta/</tt>
</td><td align="left"><p>For each toplevel HOG group we provide a
                    separate Fasta file with all protein sequences clustered
                    to it. This format is particularly suitable as starting 
                    point for a phylogenetic reconstruction of a gene tree.</p>
</td></tr>
<tr><td align="left"><tt>used_splicing_variants.txt</tt>
</td><td align="left"><p>If <tt>UseOnlyOneSplicingVariant</tt>
is activated and
                    splicing information is available, the variant which has been used 
                    for calling the orthologous relations is stored in this file.
                    The format is a tab-delimitted text file with the species in the 
                    first column and the id of the used splicing variant in the second column.</p>
</td></tr>
</tbody>
</table>
<a name="espritout"><h3>ESPRIT Output</h3>
</a><p>ESPRIT stores its output files in a directory called<tt>EspritOutput</tt>
 in your working directory. The output
            consists of three text files and one tarball. In the tarball,
            FASTA files with the MSAs of the hits ESPRIT found are stored.
             The other three files are explained in detail in Table 2.</p>
<table><caption>Table 2: Contents of the ESPRIT output files</caption><thead>
<tr><th align="left"><b>Filename</b>
</th><th align="left"><b>Contents</b>
</th></tr>
</thead>
<tbody><tr><td colspan=2 style="padding: 0px; border-bottom:1px solid black"></td></tr>
<tr><td align="left"><tt>params.txt</tt>
</td><td align="left"><p>This file is kept as a reference and 
                    contains all parameters used in the current run.</p>
</td></tr>
<tr><td align="left"><tt>hits.txt</tt>
</td><td align="left"><p>All hits found by ESPRIT are listed in this
                    file. It is a list of contigs, ordered according to their
                    position relative to the putative ortholog. Each line 
                    describes one contig, the fields are separated by tabs.
                    In the first field, the fragment pair ID is printed; the
                    next two fields contain the labels of the first and
                    second fragments found in this hit. The forth and fifth 
                    fields contain the label of the corresponding full gene
                    and its genome name. Then follows the distance difference
                    between the two fragments and the number of positions
                    between them (i.e. the gap); at last, an array is listed
                    containing the IDs of all s3 genes corresponding to this
                    hit.</p>
</td></tr>
<tr><td align="left"><tt>dubious.txt</tt>
</td><td align="left"><p>ESPRIT often detects more candidate pairs than
                    it will list in the <tt>hits.txt</tt>
 file, but not all
                    of them survive the quality check. Still, if you want to
                    see which triplets have been filtered out, have a look at <tt>dubious.txt</tt>
 where they are still listed. The file
                    format is the same as for <tt>hits.txt</tt>
.</p>
</td></tr>
</tbody>
</table>
<a name="parameters"><h1>Parameters</h1>
</a><p>All parameters for OMA and/or ESPRIT are
                set in a parameters file. There is an example file in the
                OMA installer package; we encourage you to copy this file
                into your working directory and change it to your needs.</p>
<p>The parameter file consists of two main parts:
                First, general parameters for OMA are set; see Table 3 for
                detailed explanations. Second, more specific parameters 
                that only affect the ESPRIT algorithm can be changed. 
                These parameters are explained in Table 4.
                Note that changing the ESPRIT parameters
                will not have an effect unless you set the boolean variable <tt>UseEsprit</tt>
 to <tt>true</tt>
.</p>
<table><caption>Table 3: General parameters in OMA</caption><thead>
<tr><th align="left"><b>Parameter</b>
</th><th align="left"><b>Meaning</b>
</th><th align="left"><b>Default</b>
</th></tr>
</thead>
<tbody><tr><td colspan=3 style="padding: 0px; border-bottom:1px solid black"></td></tr>
<tr><td align="left"><tt>InputDataType</tt>
</td><td align="left"><p>Type of input sequences. This can be set
                        either to 'AA' for amino acid sequences or 'DNA' for
                        nucleotide sequences</p>
</td><td align="left"><tt>AA</tt>
</td></tr>
<tr><td align="left"><tt>OutputFolder</tt>
</td><td align="left"><p>Folder to which the output is written. At
                        each run, the content of this folder will be overwritten.
                        Don't store any important files in it. The <tt>OutputFolder</tt>
must not contain any spaces.</p>
</td><td align="left"><tt>Output</tt>
</td></tr>
<tr><td align="left"><tt>ReuseCachedResults</tt>
</td><td align="left"><p>If you want to recompute everything from
                        scratch everytime the script is run, set this to <tt>false</tt>
.</p>
</td><td align="left"><tt>true</tt>
</td></tr>
<tr><td align="left"><tt>AlignBatchSize</tt>
</td><td align="left"><p>In the all-against-all phase, each genome
                        pair is split in smaller chunks of <tt>AlignBatchSize</tt>
protein comparisons. The larger this number, the longer 
                        each unit runs, and the fewer files get produced. This 
                        allows to adjust the frequency of milestone steps 
                        (e.g. in case of computer crash) or to process few but 
                        large genomes with many CPUs efficiently.</p>
</td><td align="left">1000000
</td></tr>
<tr><td align="left"><tt>MinScore</tt>
</td><td align="left"><p>Alignments which have a score lower than <tt>MinScore</tt>
 will not be considered. The scores
                        are in Gonnet PAM matrices units.</p>
</td><td align="left"><tt>181</tt>
</td></tr>
<tr><td align="left"><tt>LengthTol</tt>
</td><td align="left"><p>Length tolerance ratio. If the length of
                        the effective alignment is less than <tt>LengthTol</tt>
 * min( length(s1), length(s2) ), then 
                        the alignment is not considered.</p>
</td><td align="left"><tt>0.61</tt>
</td></tr>
<tr><td align="left"><tt>StablePairTol</tt>
</td><td align="left"><p>During the stable pair formation, if a
                        pair has a distance provable higher than another pair
                        (i.e. <tt>StablePairTol</tt>
 standard deviations
                        away) then it is discarded.</p>
</td><td align="left"><tt>1.81</tt>
</td></tr>
<tr><td align="left"><tt>VerifiedPairTol</tt>
</td><td align="left"><p>Tolerance parameter for the detection of 
                        differential gene losses using a third genome. The larger 
                        the tolerance, the liberaler the algorithm assigns 
                        orthologous relations. A detailed description is provided <a href="http://nar.oxfordjournals.org/content/34/11/3309.full">here.</a></p>
</td><td align="left"><tt>1.53</tt>
</td></tr>
<tr><td align="left"><tt>MinSeqLen</tt>
</td><td align="left"><p>Any sequence which is less than <tt>MinSeqLen</tt>
 amino acids long in regular genomes
                        is not considered.</p>
</td><td align="left"><tt>50</tt>
</td></tr>
<tr><td align="left"><tt>UseOnlyOneSplicingVariant</tt>
</td><td align="left"><p>Enables/disables the filtering on a single represetative 
                        splicing variant. If enabled, OMA selects the variant that has the most 
                        homologous matches with all other genomes. Orthology inference is then 
                        only based on this variant. If disabled, alternative splicing variants 
                        will usually be infered as paralogs.</p>
</td><td align="left"><tt>true</tt>
</td></tr>
<tr><td align="left"><tt>StableIdsForGroups</tt>
</td><td align="left"><p>Enables/disables the generation of stable identifiers for OMA groups 
                        (and Hierarchical Groups if their computation enabled). The identifier consists of 
                        a prefix to determine the type of the group ('OMA' or 'HOG'), and a subsequence
                        of the amino acid sequence uniquely present in this group. The computation of these 
                        ids might require a substantial amount of time. The ids are stored in the OrthoXML
                        files only.</p>
</td><td align="left"><tt>false</tt>
</td></tr>
<tr><td align="left"><tt>DoHierarchicalGroups</tt>
</td><td align="left"><p>Enables/disables the computation of the 
                       hierarchical groups. This steps requires substantial 
                       computating power.</p>
</td><td align="left"><tt>true</tt>
</td></tr>
<tr><td align="left"><tt>MaxTimePerLevel</tt>
</td><td align="left"><p>Define maximum amount of time (in sec) spent by 
                       the program for breaking every connected component of the 
                       orthology graph at its weakest link on a given taxonomic level. 
                       If set to a negative value, no timelimit is enforced.</p>
</td><td align="left"><tt>1200</tt>
</td></tr>
<tr><td align="left"><tt>SpeciesTree</tt>
</td><td align="left"><p>The hierarchical groups require a (partially)
                       resolved species phylogeny. With the parameter <tt>SpeciesTree</tt>
 the user can specify 
                       a phylogeny in Newick-format, or, by setting the variable 
                       to "estimate", compute a species tree based on the OMA 
                       Groups and use this one.</p>
</td><td align="left"><tt>estimate</tt>
</td></tr>
<tr><td align="left"><tt>ReachabilityCutoff</tt>
</td><td align="left"><p>The cutoff of "average reachability within 
                       two steps" defines up to what point a cluster is split 
                       into sub-clusters.</p>
</td><td align="left"><tt>0.65</tt>
</td></tr>
<tr><td align="left"><tt>WriteOutput_*</tt>
</td><td align="left"><p>Switches to disable the generation of certain 
                       output files if set to <tt>false</tt>
. For very big analyses, 
                       disabling the generation of unused output might save a 
                       substantial amount of computing time and might drastically 
                       reduce the number of produced files.</p>
</td><td align="left"><tt>true</tt>
</td></tr>
</tbody>
</table>
<table><caption>Table 4: ESPRIT parameters</caption><thead>
<tr><th align="left"><b>Parameter</b>
</th><th align="left"><b>Meaning</b>
</th><th align="left"><b>Default</b>
</th></tr>
</thead>
<tbody><tr><td colspan=3 style="padding: 0px; border-bottom:1px solid black"></td></tr>
<tr><td align="left"><tt>UseEsprit</tt>
</td><td align="left"><p>You can either set this to <tt>true</tt>
, which will enable ESPRIT and shut down the parts of
                        OMA that are not directly needed for ESPRIT, or set it
                        to <tt>false</tt>
 to make no use of ESPRIT at all.</p>
</td><td align="left"><tt>false</tt>
</td></tr>
<tr><td align="left"><tt>DistConfLevel</tt>
</td><td align="left"><p>Confidence level variable for contigs.
                        This is the parameter <tt>tol</tt>
 described in the
                        paper.</p>
</td><td align="left"><tt>2</tt>
</td></tr>
<tr><td align="left"><tt>MinProbContig</tt>
</td><td align="left"><p>Minimal proportion of genomes with which
                        contigs form many:1 BestMatches to consider that we
                        might be dealing with fragments of the same gene.
                        This is the parameter <tt>MinRefGenomes</tt>
 described in the paper, normalized by the total
                        number of reference genomes.</p>
</td><td align="left"><tt>0.4</tt>
</td></tr>
<tr><td align="left"><tt>MaxContigOverlap</tt>
</td><td align="left"><p>Maximum overlap between fragments of
                        same gene from different contigs.</p>
</td><td align="left"><tt>5</tt>
</td></tr>
<tr><td align="left"><tt>MinSeqLenContig</tt>
</td><td align="left"><p>Any sequence which is less than <tt>MinSeqLenContig</tt>
 amino acids long in contigs
                        is not considered.</p>
</td><td align="left"><tt>20</tt>
</td></tr>
<tr><td align="left"><tt>MinBestScore</tt>
</td><td align="left"><p>Minimum best score for BestMatch in
                        scaffold recognition.</p>
</td><td align="left"><tt>250</tt>
</td></tr>
</tbody>
</table>
<a name="gettinghelp"><h1>Getting help</h1>
</a><p>The preferred way to get help about OMA is via the 
            <a href="https://www.biostars.org/t/oma/">Biostars</a> community resource. Please consider asking your 
            question there, including the tags "OMA" and "orthologs".</p>
<p>If your question requires privacy, we are also 
            reachable by email at contact@omabrowser.org. </p>
<a name="license"><h1>License</h1>
</a><p>OMA standalone is licensed under the Mozilla 
Public License Version 2.0. For more info, please consult the following page:</p>
<p><a href="http://www.mozilla.org/MPL/2.0/">http://www.mozilla.org/MPL/2.0/</a></p>
<p>In a nutshell, OMA standalone is open source and 
free for commercial and non-commercial use.</p>
<p>Software distributed under the License is 
distributed on an "AS IS" basis,WITHOUT WARRANTY OF ANY KIND, either 
express or implied. See the License for the specific language governing 
rights and limitations under the License.</p>
</div></body></html>

